perceval/backend.py

```diff
@@ -207,6 +236,42 @@ def filter_classified_data(self, item):
 
         return item
 
+    def search_fields(self, item):
+        """Add search fields to an item.
+
+        It adds the values of the fields defined in `SEARCH_FIELDS` class attribute with
+        their corresponding keys.
+
+        :param item: the item to extract the search fields values
+
+        :returns: a dict of search fields
+        """
+        item_uuid = uuid(self.origin, self.metadata_id(item))
+
+        logger.debug("Adding search fields to item %s", item_uuid)
+
+        logger.debug("Adding default `item_id` search field to item %s", item_uuid)
+        search_fields = {
+            DEFAULT_SEARCH_TERM: self.metadata_id(item)
+        }
+
+        logger.debug("Adding extra search fields to item %s", item_uuid)
+        for sf in self.EXTRA_SEARCH_FIELDS:
+            try:
+                search_field = self.EXTRA_SEARCH_FIELDS[sf]
+                field_value = _find_value_from_nested_dict(item, search_field)
+                search_fields[sf] = field_value
+            except KeyError:
+                logger.debug("Extra search field '%s' not found for item %s; field ignored",
```

Comment:
I guess this cases should be not common. I think this should be a warning.

Reply:
Thanks @sduenas! 
Sure! Should I updated also this line: https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backend.py#L203 ?

Reply:
No I think that case is different. From my point of view, I think all the fields included in the `search_fields` should be mandatory and available on all items. In the case of `filtered_classified` those fields may appear or not, cause depending on how you get the data (for example, using authentication in bugzilla you get the email addresses of the users) you can have those values or not.

Reply:
PR updated